python/ml4ir/applications/classification/model/classification_model.py

```diff
@@ -140,19 +180,31 @@ def predict(
             outfile = os.path.join(logs_dir, RelevanceModelConstants.MODEL_PREDICTIONS_CSV_FILE)
             # Delete file if it exists
             self.file_io.rm_file(outfile)
+        predictions_df = self._create_prediction_dataframe(logging_frequency,
+                                                           test_dataset)
+        predictions_ = np.squeeze(self.model.predict(test_dataset))
+        # Below, avoid doing predictions.tolist() as it explodes the memory
+        predictions_df[self.output_name] = [x for x in predictions_]
```

Comment:
this is also very important. `.tolist()` increases the memory a lot.. Some examples here: https://stackoverflow.com/questions/43122012/why-is-nparray-tolist-taking-this-much-space

Reply:
`tolist` is more readable. But the comprehension maintains the inner arrays as np.array that saves memory.

Reply:
You're right. I forgot that these are 2d lists